cpp/src/cider/type/plan/UnaryExpr.cpp

```diff
@@ -337,11 +393,19 @@ JITExprValue& UOper::codegenCast(CodegenContext& context, Analyzer::Expr* operan
                                   .params_vector = {ptr_and_len.get()}});
     return set_expr_value(operand_val.getNull(), ret_len, ret_ptr);
   } else if (operand_ti.is_string()) {
-    auto str_val = VarSizeJITExprValue(operand->codegen(context));
-    return set_expr_value(
-        str_val.getNull(),
-        codegenCastStringToNumeric(
-            context, str_val.getValue(), str_val.getLength(), target_ti));
+    if (Constant* consant_expr = dynamic_cast<Constant*>(operand); consant_expr) {
+      return set_expr_value(
+          func.createLiteral(JITTypeTag::BOOL,
+                             consant_expr->get_is_null() ? true : false),
```

Comment:
seems the target is `int,date,timestamp` etc, can't be null?

Reply:
Yes

Reply:
then we just need to create literal `false`